src/OpenDiffix.Core/Analyzer.fs

```diff
@@ -42,11 +42,19 @@ let private mapFunctionExpression rangeColumns fn parsedArgs =
   let mapAids parsedAids =
     parsedAids |> List.map (mapExpression rangeColumns) |> ListExpr
 
+  let mapAvg aggregateArgs parsedArgs sumFunction countFunction =
+    let sum = mapFunctionExpression rangeColumns (AggregateFunction(sumFunction, aggregateArgs)) parsedArgs
+    let count = mapFunctionExpression rangeColumns (AggregateFunction(countFunction, aggregateArgs)) parsedArgs
+    let castSum = FunctionExpr(ScalarFunction ScalarFunction.Cast, [ sum; Constant(String "real") ])
```

Comment:
Is the extra cast dropped later if it's redundant?

Reply:
Correction, it does gets dropped, I'm blind. For some reason it doesn't for `cast(1.0, 'real')`, I'll investigate further.

But I guess we're good here, in the context of `avg`?

Reply:
> For some reason it doesn't for cast(1.0, 'real')

Should be folded by the compiler/normalizer. Worth fixing as it's no-op.

Reply:
> But I guess we're good here, in the context of avg?

Yes.

> For some reason it doesn't for cast(1.0, 'real')

Interesting, worth investigating and fixing.

Reply:
Ok, I've got it, will file a follow up PR to fix